device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttMessaging.java

```diff
@@ -167,10 +168,34 @@ public void send(Message message) throws IOException
             throw new IOException("Message cannot be null");
         }
 
+        MessageProperty[] messageProperties = message.getProperties();
+        String messagePublishTopic;
```

Comment:
Thank you for your change. This looks good and is much needed. Can you please also add a test case around this in device\iot-device-client\src\test\java\tests\unit\com\microsoft\azure\sdk\iot\device\transport\mqtt\MqttMessagingTest.java ?